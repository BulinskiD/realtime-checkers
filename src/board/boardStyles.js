import styled, {css} from "styled-components";

export const BoardContainer = styled.div`
        margin: 0 auto;
        height: 90vh;
        width: 90vh;
        display: grid;
        grid-auto-flow: row;
        grid-template-columns: 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5%;
        grid-template-rows: 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5%;
        justify-items: stretch;
    `;

const CheckerItem = styled.span`
        width: 60%;
        height: 60%;
        border-radius: 100%;   
        outline-offset: 6px;
        transition: all .2s;     
    `;

export const ColorChecker = styled(CheckerItem)`
        background-color: ${props => props.color === 'white' ? 'lightgray' : 'black'};
        ${props => (props.nextMove && props.hover) && css`
           &:hover {
            outline: 2px solid ${props => props.color === 'white' ? 'lightgray' : 'black'};
            }
        `}
    `;

export const SelectedChecker = styled(CheckerItem)`
         outline: 2px solid black;
         background-color: ${props => props.color === 'black'? 'black' : 'lightgray'}
    `;

export const PoleContainer = styled.div`
        border: solid 1px black;
        display: flex;
        justify-content: center;
        align-items: center;
    `;

export const ActivePlace = styled.div`
        background-color: red;
        cursor: pointer;
        width: 100%;
        height: 100%;
    `;