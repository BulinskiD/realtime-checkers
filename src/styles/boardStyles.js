import styled, { css } from "styled-components";

export const FlexContainer = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const PlayersManagerContainer = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 25%;
`;

export const PlayerHolderTitle = styled.h4`
  font-size: 1.4em;
  font-weight: 300;
`;

export const PlayerHolder = styled.div`
  display: flex;
  flex-direction: column;
  padding: 5%;
  padding-top: 10%;
  padding-bottom: 10%;
  background-color: rgba(0, 0, 0, 0.05);
  border: 1px solid lightgray;
  box-shadow: 5px 5px 10px 0px rgba(0, 0, 0, 0.15);
`;

export const PlayerItem = styled.span`
  padding: 5%;
  background-color: ${props =>
    props.active ? "#cc3300" : "rgba(204, 51, 0, .4)"};
  color: white;
  position: relative;
  border: 2px rgb(1, 37, 96, 0.5) solid;
  &:not(:first-child) {
    margin-top: 5%;
  }

  &::after {
    position: absolute;
    left: -14px;
    top: 50%;
    transform: translateY(-50%);
    z-index: 5;
    display: block;
    content: " ";
    width: 10px;
    height: 10px;
    background-color: ${props => props.color};
  }
`;

export const BoardContainer = styled.div`
  height: 80vh;
  width: 80vh;
  display: grid;
  grid-auto-flow: row;
  grid-template-columns: 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5%;
  grid-template-rows: 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5% 12.5%;
  justify-items: stretch;
`;

const CheckerItem = styled.span`
  width: 60%;
  height: 60%;
  border-radius: 100%;
  outline-offset: 6px;
  transition: all 0.2s;
  position: relative;
`;

export const ColorChecker = styled(CheckerItem)`
  background-image: ${props =>
    props.color === "white"
      ? "radial-gradient(lightgray, gray)"
      : "radial-gradient(#333, black)"};
  ${props =>
    props.isKing &&
    css`
      &::after {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        display: block;
        content: "";
        width: 20px;
        height: 20px;
        background: tomato;
      }
    `}
  ${props =>
    props.nextMove &&
    props.hover &&
    css`
      &:hover {
        outline: 2px solid
          ${props => (props.color === "white" ? "lightgray" : "black")};
      }
    `}
`;

export const SelectedChecker = styled(CheckerItem)`
  outline: 2px solid
    ${props => (props.color === "black" ? "black" : "lightgray")};
  background-color: ${props =>
    props.color === "black" ? "black" : "lightgray"};
  ${props =>
    props.isKing &&
    css`
      &::after {
        position: absolute;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        display: block;
        content: "";
        width: 20px;
        height: 20px;
        background: tomato;
      }
    `}
`;

export const PoleContainer = styled.div`
  border: solid 1px black;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${props =>
    (props.col % 2 === 0 && props.row % 2 === 0) ||
    (props.col % 2 === 1 && props.row % 2 === 1)
      ? "#993300"
      : "#ffffcc"};
`;

export const ActivePlace = styled.div`
  background-color: red;
  cursor: pointer;
  width: 100%;
  height: 100%;
`;

export const TimeIndicator = styled.div`
  background-image: linear-gradient(
    to right,
    rgba(204, 51, 0, 0.4),
    rgba(204, 51, 0, 1)
  );
  height: 3vh;
  width: ${props => props.percentage}px;
`;
