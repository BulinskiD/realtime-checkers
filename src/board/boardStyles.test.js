import React from 'react';
import {mount} from 'enzyme';
import { ColorChecker, SelectedChecker, PlayerItem} from "./boardStyles";

describe("ColorChecker", () => {
   it('should render white component for given props', () => {
       const component = mount(<ColorChecker color="white" isKing={false} nextMove={false} hover={false} />);
       expect(component).toMatchSnapshot();
   });

    it('should render black component for given props', () => {
        const component = mount(<ColorChecker color="black" isKing={false} nextMove={false} hover={false} />);
        expect(component).toMatchSnapshot();
    });

    it('should add hover pseudo class for given props', () => {
        const component = mount(<ColorChecker color="black" isKing={false} nextMove={true} hover={true} />);
        expect(component).toMatchSnapshot();
    });

    it('should render component with king markup', () => {
        const component = mount(<ColorChecker color="black" isKing={true} nextMove={true} hover={true} />);
        expect(component).toMatchSnapshot();
    });
});

describe('SelectedChecker', () => {
   it('should render black selectedChecker with king markup for given props', () => {
        const component = mount(<SelectedChecker color="black" isKing={true} />);
        expect(component).toMatchSnapshot();
    });

    it('should render white selectedChecker with king markup for given props', () => {
        const component = mount(<SelectedChecker color="white" isKing={true} />);
        expect(component).toMatchSnapshot();
    });

    it('should render white selectedChecker without king markup for given props', () => {
        const component = mount(<SelectedChecker color="white" isKing={false} />);
        expect(component).toMatchSnapshot();
    });

    it('should render black selectedChecker without king markup for given props', () => {
        const component = mount(<SelectedChecker color="black" isKing={false} />);
        expect(component).toMatchSnapshot();
    });
});

describe('PlayerItem', () => {
   it('should render darker blue color for active player', () => {
       const component = mount(<PlayerItem active={true} color="black" />);
       expect(component).toMatchSnapshot();
   })

   it('should render dimmer blue color for inactive player', () => {
       const component = mount(<PlayerItem active={false} color='white' />);
       expect(component).toMatchSnapshot();
   });
});