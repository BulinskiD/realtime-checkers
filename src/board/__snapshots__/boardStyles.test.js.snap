// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorChecker should add hover pseudo class for given props 1`] = `
<Styled(styled.span)
  color="black"
  hover={true}
  isKing={false}
  nextMove={true}
>
  <span
    className="sc-EHOje sc-bZQynM kqXkkz"
    color="black"
  />
</Styled(styled.span)>
`;

exports[`ColorChecker should render black component for given props 1`] = `
<Styled(styled.span)
  color="black"
  hover={false}
  isKing={false}
  nextMove={false}
>
  <span
    className="sc-EHOje sc-bZQynM bWCycq"
    color="black"
  />
</Styled(styled.span)>
`;

exports[`ColorChecker should render component with king markup 1`] = `
<Styled(styled.span)
  color="black"
  hover={true}
  isKing={true}
  nextMove={true}
>
  <span
    className="sc-EHOje sc-bZQynM jxatWb"
    color="black"
  />
</Styled(styled.span)>
`;

exports[`ColorChecker should render white component for given props 1`] = `
<Styled(styled.span)
  color="white"
  hover={false}
  isKing={false}
  nextMove={false}
>
  <span
    className="sc-EHOje sc-bZQynM GyDiX"
    color="white"
  />
</Styled(styled.span)>
`;

exports[`PlayerItem should render darker blue color for active player 1`] = `
<styled.span
  active={true}
  color="black"
>
  <span
    className="sc-bxivhb gFEUqD"
    color="black"
  />
</styled.span>
`;

exports[`PlayerItem should render dimmer blue color for inactive player 1`] = `
<styled.span
  active={false}
  color="white"
>
  <span
    className="sc-bxivhb bWMQot"
    color="white"
  />
</styled.span>
`;

exports[`SelectedChecker should render black selectedChecker with king markup for given props 1`] = `
<Styled(styled.span)
  color="black"
  isKing={true}
>
  <span
    className="sc-EHOje sc-gzVnrw jQQcLS"
    color="black"
  />
</Styled(styled.span)>
`;

exports[`SelectedChecker should render black selectedChecker without king markup for given props 1`] = `
<Styled(styled.span)
  color="black"
  isKing={false}
>
  <span
    className="sc-EHOje sc-gzVnrw buauLl"
    color="black"
  />
</Styled(styled.span)>
`;

exports[`SelectedChecker should render white selectedChecker with king markup for given props 1`] = `
<Styled(styled.span)
  color="white"
  isKing={true}
>
  <span
    className="sc-EHOje sc-gzVnrw cANMAy"
    color="white"
  />
</Styled(styled.span)>
`;

exports[`SelectedChecker should render white selectedChecker without king markup for given props 1`] = `
<Styled(styled.span)
  color="white"
  isKing={false}
>
  <span
    className="sc-EHOje sc-gzVnrw gTWeLk"
    color="white"
  />
</Styled(styled.span)>
`;
